src/drv/stm32cube/bsp_i2c_master.c

```diff
@@ -261,8 +301,7 @@ void bsp_i2c_read_ack(bsp_dev_i2c_t dev_num, bool enable_ack)
 
 	i2c_sw_delay();
 
-	set_scl_float();
-	i2c_sw_delay();
+	i2c_master_set_scl_float_and_wait_ready();
```

Comment:
The prototype of that function shall be changed from void to bsp_status_t 
Check the return code and return it

Reply:
same as the other update.

Reply:
i have made a bit change to handle the error inside i2c mode in a unified place, hence they are tracked together now. resolved as dup for now, so we have a single place for discussion.